eng/packages/http-client-csharp-mgmt/generator/Azure.Generator.Management/src/Providers/TagMethodProviders/BaseTagMethodProvider.cs

```diff
@@ -188,18 +189,24 @@ protected static MethodBodyStatement GetOriginalTagsStatement(
         protected MethodBodyStatement UpdateResourceStatement(
             VariableExpression dataVar,
             ParameterProvider cancellationTokenParam,
+            MethodProvider updateMethod,
             out VariableExpression resultVar)
         {
-            var updateMethod = _isAsync ? "UpdateAsync" : "Update";
+            var updateMethodName = _isAsync ? "UpdateAsync" : "Update";
+            var parameters = new List<ValueExpression>();
+            var isLro = updateMethod.Signature.ReturnType?.FrameworkType == typeof(ArmOperation<>);
```

Comment:
We should not check isAsync from the return type.
Instead, we could expose _isLongRunningOperation as property in https://github.com/Azure/azure-sdk-for-net/blob/df728e791f78dfdca1a095cc8333ec0c084827fc/eng/packages/http-client-csharp-mgmt/generator/Azure.Generator.Management/src/Providers/OperationMethodProviders/ResourceOperationMethodProvider.cs#L44
Then, we can fetch it via _updateMethod, there is no need to add this as input parameter,